alacritty_terminal/src/config/window.rs

```diff
@@ -117,3 +141,45 @@ impl Dimensions {
         self.columns.0 as u32
     }
 }
+
+/// Window class
+#[derive(Clone, Default, Debug, Eq, PartialEq)]
+pub struct Class {
+    pub instance: Option<String>,
+    pub class: Option<String>
+}
+
+impl<'de> Deserialize<'de> for Class {
+    fn deserialize<D>(deserializer: D) -> Result<Class, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = serde_yaml::Value::deserialize(deserializer)?;
+
+        match value {
+            Value::Null => Ok(Default::default()),
+            Value::String(val) => Ok(Class { instance: Some(val), class: None }),
+            Value::Sequence(mut vals) => {
```

Comment:
I feel like we might be making it harder on ourselves with parsing here by using an array. Instead, maybe we can consider doing something similar to what `shell` allows? So both of the following would be allowed:

```yml
class: Alacritty
```

```yml
class:
  instance: Alacritty
  class: Alacritty
```

I think it would be nice to find a better name for class->class, since duplication here will inevitably lead to confusion, but I'm not sure if there's a standardized name for the part of the class which should always stay the same across instances.

>  A string that names the general class of applications to which the client that owns this window belongs. Resources that are specified by class apply to all applications that have the same class name. Class names are specified by the application writer. Examples of commonly used class names include: "Emacs", "XTerm", "XClock", "XLoad", and so on. 

Based on this documentation, maybe `general` is an option?

Reply:
The official X11 structure is:

```c
typedef struct {
    char *res_name;
    char *res_class; 
} XClassHint; 
```

So I could change `instance` to `name`, but the class/class nomenclature is standard.

Reply:
I'm aware that by default this is just the 'window class', however if we want to have a reasonable configuration file and be able to communicate reasonably about our own code, we need to find good words to identify the struct and the struct's fields.

I think `instance` is great already, this is often what it's referred to in the documentation too (we don't need to stick to the exact struct). So the only remaining question is struct name and general class.

I think naming the struct `XClassHint would be more confusing than helpful, we should name this closely similar to the config field and I think the most reasonable for the user there would be the `class` field identifier. So we need to find a good name for the 'opposite' of instance. And that's what I'd propose `general` for. The general class is used for all instances and that's the most reasonable naming I could come up with looking at the documentation of X.